File: Array/Two Sum/solutionUsingHashmap.cpp
class Solution {
   public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numsMap;
        int size = nums.size();

        for (int i = 0; i < size; i++) {
            numsMap[nums[i]] = i;
        }

        for (int i = 0; i < size; i++) {
            int numNeeded = target - nums[i];

            if (numsMap.count(numNeeded) && numsMap[numNeeded] != i) {
                return {i, numsMap[numNeeded]};
            }
        }
        return {};
    }
};

File: Array/Two Sum/solutionUsingHashmap.go
package main

func twoSum(nums []int, target int) []int {
	var result []int
	hashMap := make(map[int]int)

	for i, num := range nums {
		found := target - num
		if val, ok := hashMap[found]; ok {
			result = append(result, i, val)
		}
		hashMap[num] = i
	}

	return result
}
